kpi/tests/api/v2/test_api_submissions.py

```diff
@@ -215,6 +215,20 @@ def test_list_submissions_anonymous_asset_publicly_shared(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.asset.remove_perm(anonymous_user, 'view_submissions')
 
+    def test_list_submissions_authenticated_asset_publicly_shared(self):
+        # https://github.com/kobotoolbox/kpi/issues/2698
+        self._log_in_as_another_user()
+
+        # give the poor schmuck their own asset; needed to expose flaw in
+        # `ObjectPermissionMixin.__get_object_permissions()`
+        Asset.objects.create(name='i own it', owner=self.anotheruser)
+
+        anonymous_user = get_anonymous_user()
+        self.asset.assign_perm(anonymous_user, 'view_submissions')
+        response = self.client.get(self.submission_url, {"format": "json"})
```

Comment:
I must miss something. I don't understand why we try to fetch the submission with the asset's owner. 
Owners receive `view_submissions` by default, don't they?
I thought the purpose of the PR to let other authenticated users to see publicly shared data. 

For example: 
jnm's asset is publicly shared.
noliveleger's doesn't have any permission on this asset, but thanks to this fix, is able to view the data.

Reply:
`self.asset` is owned by `someuser`, not `anotheruser`

Reply:
I've attempted to improve the documentation. Does it make sense?

Reply:
I don't see any other documentation since yesterday?